syntax = "proto3";

package huddle_chat;

import "google/protobuf/timestamp.proto";


message Guild {
  int64 guild_id = 1;
  string name = 2;
  google.protobuf.Timestamp created_at = 3;
  string icon = 4;
  string description = 5;
  bool is_owner = 6;
  bool has_unread = 7;
}

message GuildsByUserIdRequest {
  int64 user_id = 1;
}

message GuildsByUserIdResponse {
  repeated Guild guilds = 1;
}

message CreateGuildRequest {
  int64 user_id = 1;
  string name = 2;
  string icon = 3;
  string description = 4;
}

message CreateGuildResponse {
  Guild guild = 1;
}

message DeleteGuildRequest {
  int64 user_id = 1;
  int64 guild_id = 2;
}

message DeleteGuildResponse {
  Guild guild = 1;
}

service GuildService {
  rpc GetGuildsByUserId(GuildsByUserIdRequest) returns (GuildsByUserIdResponse) {}
  rpc CreateGuild(CreateGuildRequest) returns (CreateGuildResponse) {}
  rpc DeleteGuild(DeleteGuildRequest) returns (DeleteGuildResponse) {}
}
